### PR TITLE
fix(DB/Quest): Port quests regarding Corki from Trinity

### PR DESCRIPTION
**Pull request by v-mstrs**
https://github.com/TrinityCore/TrinityCore/commit/fd66e291b5595635411116b5b0946b476719a2ab
Co-authored-by: Killyana <Killyana@users.noreply.github.com>

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14612
- Closes https://github.com/chromiecraft/chromiecraft/issues/4655

## Tests Performed:

- Tested

## How to Test the Changes:

1. .tele nagrand, .go xyz -2588.2 6277.3 25.30
2. .modify rep 978 10000
3. Start quest chain

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Please try to include the linked ChromieCraft issue in the issues addressed section to close that as well.

File: data/sql/updates/pending_db_world/rev_1682394052134857558.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+UPDATE `creature_template` SET `AIName`= 'SmartAI',`ScriptName`='' WHERE `entry` IN (18445,18369,20812);
+UPDATE `gameobject_template` SET `Data2`=300000, `AIName`='SmartGameObjectAI', `ScriptName`='' WHERE `entry` IN (182521,182349,182350);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1682394052134857558.sql:3]: Just some formatting here

```suggestion
UPDATE `creature_template` SET `AIName`= 'SmartAI', `ScriptName`='' WHERE `entry` IN (18445,18369,20812);
UPDATE `gameobject_template` SET `Data2`=300000, `AIName`='SmartGameObjectAI', `ScriptName`='' WHERE `entry` IN (182521,182349,182350);
```

File: data/sql/updates/pending_db_world/rev_1682394052134857558.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(182521, 1, 0 ,1,70, 0, 100, 0, 2, 0, 0,0,64,1,0,0,0,0,0,16,0,0,0,0, 0, 0, 0, 'Corkis Prison - On State Changed - Store Targetlist'),
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1682394052134857558.sql:6]: Missing DELETE queries here. You can find them in the commit you linked

File: data/sql/updates/pending_db_world/rev_1682394052134857558.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+(182350, 1, 1 ,2,61, 0, 100, 0, 0, 0, 0,0,100,1,0,0,0,0,0,19,20812,0,0,0, 0, 0, 0, 'Corkis Prison - Script - Send Targetlist'),
+(182350, 1, 2 ,0,61, 0, 100, 0, 0, 0, 0,0,45,1,1,0,0,0,0,19,20812,0,0,0, 0, 0, 0, 'Corkis Prison - Script - Set Data'),
+(20812, 0, 0 ,0,38, 0, 100, 0, 1, 1, 0,0,80,2081200,0,0,0,0,0,1,0,0,0,0, 0, 0, 0, 'Corki - On Data Set - Action list'),
+(2081200, 9, 0 , 0, 0, 0, 100, 0, 0, 0, 0,0,1,0,0,0,0,0,0,12,1,0,0,0, 0, 0, 0, 'Corki - Action list - Say'),
+(2081200, 9, 1 , 0, 0, 0, 100, 0, 0, 0, 0,0,33,20812,0,0,0,0,0,12,1,0,0,0, 0, 0, 0, 'Corki - Action list - Kill Credit'),
+(2081200, 9, 2 , 0, 0, 0, 100, 0, 4000, 4000, 0,0,69,0,0,0,0,0,0,8,0,0,0,-1001.022, 8113.366, -95.849, 0.352908, 'Corki - Action list - Move Foward'),
+(2081200, 9, 3 , 0, 0, 0, 100, 0, 4000, 4000, 0,0,41,0,0,0,0,0,0,1,0,0,0,0, 0, 0, 0, 'Corki - Action list - Despawn after 5 seconds');
+
+UPDATE `creature` SET `position_x`=-2563.89,`position_y`=6288.29,`position_z`=15.295,`orientation`=5.23599 WHERE `id1` = (18369);
+UPDATE `creature` SET `position_x`=-918.143,`position_y`=8663.94,`position_z`=172.542,`orientation`=0.523599 WHERE `id1` = (18445);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1682394052134857558.sql:32]: Try to always include the guid as well when possible, unless you're absolutely certain of what you're doing

File: data/sql/updates/pending_db_world/rev_1682394052134857558.sql [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+DELETE FROM `creature_questender`  WHERE `id` IN (18369,18445);
+INSERT INTO `creature_questender` (`id`, `quest`) VALUES
+(18445, 9954);
+
+DELETE FROM `creature_queststarter`  WHERE `id` IN (18369,18445);
+INSERT INTO `creature_queststarter` (`id`, `quest`) VALUES
+(18369, 9923),
+(18445, 9955);
+-
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1682394052134857558.sql:42]: Generating errors 😛 
```suggestion
```

**Review by Gultask — CHANGES_REQUESTED** [review]
One last thing: You need to add the original author, Killyana, as a co-author.
The method varies with what you're using to publish PRs

File: data/sql/updates/pending_db_world/rev_1682394052134857558.sql (hunk 2 above)
Comment (v-mstrs) [reply data/sql/updates/pending_db_world/rev_1682394052134857558.sql:5]: Why is is necessary?

Comment (heyitsbench) [reply data/sql/updates/pending_db_world/rev_1682394052134857558.sql:5]: > Why is it necessary?

To make re-runs of queries safe, it avoids duplicate entries.

**Comment by Gultask** [comment]
Configure OS is not an important error, we just need to re-run the job later, don't worry

**Comment by GastonCK** [comment]
Tested, quests and their text lines look ok, I think he's still missing the dialogue lines he says while he's caged.

First quest
WoWScrnShot_070123_132943

Second quest
WoWScrnShot_070123_133350

Third quest
WoWScrnShot_070123_133934

Stayed with him for some time, he did not say any of the quotes he's got on wowhead.
WoWScrnShot_070123_135507
image

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**